### PR TITLE
ENH: fast stochastic oscillator added

### PR DESCRIPTION
**Pull request by jonnybazookatone**
A fast stochastic oscillator has been added to the technical factors. This is the simplest of the stochastic oscillators, and can be used to build the others.

Tests have been added that compare against the values expected from that of ta-lib `STOCHF`. They are more basic than those used in BollingerBands.

Can't wait for suggestions ;-).

### CODE REVIEW

**Comment by llllllllll** [comment]
It appears there is 1 style violation causing the ci to fail:
```
tests\pipeline\test_technical.py:202:80: E501 line too long (83 > 79 characters)
```

**Comment by jonnybazookatone** [comment]
Thanks, flake8 is the bane of my existence.

File: zipline/pipeline/factors/technical.py [hunk 1]
```diff
@@ -739,3 +742,59 @@ def compute(self, today, assets, out, close, k):
...
+    -------
+    out.K: %K oscillator
+    out.D: %D oscillator, which is %K averaged over the last D_period days
+    """
+    params = ('D_period', )
+    inputs = (USEquityPricing.close, USEquityPricing.low, USEquityPricing.high)
+    window_length = 14
+    outputs = 'K', 'D'
+
+    def compute(self, today, assets, out, closes, lows, highs, D_period=3):
```
Comment (ssanderson) [inline zipline/pipeline/factors/technical.py]: This default won't ever actually be used the way the the `params` machinery works right now.  I'd probably just make it required.

**Comment by ssanderson** [comment]
Changes here look good to me.  The branch has a merge conflict because everyone working on technical indicators at the sprint edited the bottom of the same file.  @jonnybazookatone are you comfortable rebasing this and fixing the conflict?

**Comment by jonnybazookatone** [comment]
Yeah, no problem, I'll do it by the end of today, or early tomorrow.

**Comment by jonnybazookatone** [comment]
I borked on rebase, or it's a problem with blaze.git? I see some other PRs also failed with the same error -- coincidence?

**Comment by llllllllll** [comment]
This was my doing, I just pushed a fix, sorry but can you rebase against master again?

**Comment by ssanderson** [comment]
Merged.  Thanks @jonnybazookatone!

**Pull request merged**